UTF-8 support

user185953 opened the pull request:
Based on https://github.com/cisiqo/PyWBXML/commit/770ad4cfd18d631f889d730d8b4c4903a6fb026e
The "errors='backslashreplace'" bit is there, because in mitmproxy the priority is seeing every byte

user185953 committed:
UTF-8 support

Based on https://github.com/cisiqo/PyWBXML/commit/770ad4cfd18d631f889d730d8b4c4903a6fb026e

user185953 commented:
WBXML fields are binary in general.
Binary data with the highest bit set are corrupted by PyWBXMLDecoder.
This patch changes the mode of corruption to preserve UTF-8 and escape non-UTF-8

iragsdale commented:
This looks pretty good to me @davidpshaw. This is basically how we do it in Boxer.

user185953 commented:
I apologize for the noise. Commits above make no changes. I'll try to clean up with a force push shortly

user185953 committed:
Split handling of Opaque v.s. String

Opaque nodes contain arbitrary bytes - hex() them to keep output XML valid
String nodes must contain valid UTF-8 - enforce that by errors='backslashreplace'
bytes.hex() is available since python 3.5

user185953 commented:
Thank you for the review, @iragsdale. The other commit I just pushed will be trickier.
Bytes().hex() looks odd, CDATA with hex-encoded data is probably not right and output format changes.
The general idea, however, looks OK?

davidpshaw commented:
@user185953 still need to put up a PR into mitmproxy -- this project isn't brought in as a submodule there, it's duplicated.
